client/my-sites/earn/ads/payments.jsx

```diff
@@ -1,41 +1,22 @@
 import { Badge, Card } from '@automattic/components';
 import classNames from 'classnames';
-import { localize } from 'i18n-calypso';
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import { useTranslate } from 'i18n-calypso';
 import QueryWordadsPayments from 'calypso/components/data/query-wordads-payments';
 import QueryWordadsSettings from 'calypso/components/data/query-wordads-settings';
 import Notice from 'calypso/components/notice';
 import { CALYPSO_CONTACT } from 'calypso/lib/url/support';
+import { useSelector } from 'calypso/state';
 import { getWordadsSettings } from 'calypso/state/selectors/get-wordads-settings';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { getWordAdsPayments } from 'calypso/state/wordads/payments/selectors';
 
-class WordAdsPayments extends Component {
-	static propTypes = {
-		siteId: PropTypes.number,
-		payments: PropTypes.arrayOf(
-			PropTypes.shape( {
-				id: PropTypes.number.isRequired,
-				paymentDate: PropTypes.string,
-				amount: PropTypes.string,
-				status: PropTypes.string,
-				paypalEmail: PropTypes.string.isRequired,
-				description: PropTypes.string,
-			} )
-		).isRequired,
-	};
+const WordAdsPayments = ( { numberFormat } ) => {
+	const translate = useTranslate();
+	const siteId = useSelector( ( state ) => getSelectedSiteId( state ) );
+	const payments = useSelector( ( state ) => getWordAdsPayments( state, siteId ) );
+	const wordAdsSettings = useSelector( ( state ) => getWordadsSettings( state, siteId ) );
 
-	checkSize( obj ) {
-		if ( ! obj ) {
-			return 0;
-		}
-
-		return Object.keys( obj ).length;
-	}
```

Comment:
The checkSize method here doesn't appear to be used in this component, so I've removed it.